docs/developer/meta/config-models.md

```diff
@@ -26,25 +26,22 @@ foo
 
 There are 2 possible models:
 
-- `SharedConfig` (ID: `shared`) that corresponds to the `init_config` section
 - `InstanceConfig` (ID: `instance`) that corresponds to a check's entry in the `instances` section
+- `SharedConfig` (ID: `shared`) that corresponds to the `init_config` section that is shared by all instances
 
 All models are defined in `<ID>.py` and are available for import directly under `config_models`.
 
 ## Default values
 
 The default values for optional settings are populated in `defaults.py` and are derived from the
-[value](../meta/config-specs.md#values) property of config spec options.
-
-The precedence is:
-
-1. the `default` key
-2. the `example` key, if it appears to represent a real value rather than an illustrative example and the `type` is a primitive
-3. the default value of the `type` e.g. `string` -> `str()`, `object` -> `dict()`, etc.
+[value](config-specs.md#values) property of config spec options. The precedence is the `default` key
+followed by the `example` key (if it appears to represent a real value rather than an illustrative example
+and the `type` is a primitive). In all other cases the default will be `None` and there will be no default
```

Comment:
```suggestion
and the `type` is a primitive). In all other cases, the default is `None`. There is no default
```

Reply:
This isn't quite right because the intent is to express that there is no function when there is no default

Reply:
Oh, I see, i misunderstood. Let me edit again. Thanks for explaining!